### DISCUSSION TITLE
Change `trim` to require its 2nd argument

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#68] [Ideas]
The implicit behavior could be misinterpreted as doing exactly what `String.prototype.trim` does or just confuse someone who isn't familiar with Radashi. Most of the time, if you're trimming `" "` characters, you're also trimming other whitespace (`\r`, `\t`, `\n`).

The bottom line is: **Explicit is better.**

**Comment by localusercamp** [comment]
Radashi `trim` function is an absolute mess =D.

First of all it could accept null and undefined as first argument which makes this function also a typecasting function which is wierd.
```ts
trim(undefined) // ''
```

Second thing is that it does not behave like trim in all other languages/libraries. This is related to what you are talking about. The second argument must be `" \r\t\n"` by default or have no default at all.

In my opinion default to `" \r\t\n"` is better, because it's the same as native JS trim and also looks more consistent with other languages/libraries.
